x/nexus/keeper/msg_dispatcher.go

```diff
@@ -0,0 +1,101 @@
+package keeper
+
+import (
+	"encoding/json"
+	"fmt"
+
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	axelarnet "github.com/axelarnetwork/axelar-core/x/axelarnet/exported"
+	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	"github.com/axelarnetwork/axelar-core/x/nexus/types"
+	tss "github.com/axelarnetwork/axelar-core/x/tss/exported"
+)
+
+var _ wasmkeeper.Messenger = (*Messenger)(nil)
+
+type message struct {
+	SenderChain      exported.ChainName `json:"sender_chain"`
+	SenderAddress    string             `json:"sender_address"`
+	RecipientChain   exported.ChainName `json:"recipient_chain"`
+	RecipientAddress string             `json:"recipient_address"`
+	PayloadHash      []byte             `json:"payload_hash"`
+	SourceTxID       []byte             `json:"source_tx_id"`
+	SourceTxIndex    uint64             `json:"source_tx_index"`
+}
+
+type request = []message
+
+type Messenger struct {
+	types.Nexus
+}
+
+// NewMessenger returns a new Messenger
+func NewMessenger(nexus types.Nexus) Messenger {
+	return Messenger{nexus}
+}
+
+// DispatchMsg decodes the messages from the cosmowasm connection router and routes them to the nexus module if possible
+func (m Messenger) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddress, _ string, msg wasmvmtypes.CosmosMsg) (events []sdk.Event, data [][]byte, err error) {
```

Comment:
where is this being called from?

Reply:
Nowhere at the moment. There will be another follow-up PR to hook this in.

Reply:
Actually, since it's just a one-liner, I added it to this PR.